driver-mapping/src/main/java/com/datastax/driver/mapping/ReflectionUtils.java

```diff
@@ -15,26 +15,179 @@
+        }
+        Map<String, PropertyDescriptor> properties = scanProperties(baseClass);
+        for (Map.Entry<String, PropertyDescriptor> entry : properties.entrySet()) {
+            Object[] value = fieldsAndProperties.get(entry.getKey());
+            if (value == null)
+                fieldsAndProperties.put(entry.getKey(), new Object[]{null, entry.getValue()});
+            else value[1] = entry.getValue();
+        }
+        return fieldsAndProperties;
+    }
+
+    private static <T> Map<String, Field> scanFields(Class<T> baseClass) {
+        HashMap<String, Field> fields = new HashMap<String, Field>();
+        for (Class<?> clazz = baseClass; !clazz.equals(Object.class); clazz = clazz.getSuperclass()) {
+            for (Field field : clazz.getDeclaredFields()) {
+                if (field.getName().equals("class") || field.isSynthetic() || Modifier.isStatic(field.getModifiers()))
+                    continue;
+                // never override a more specific field masking another one declared in a superclass
+                if (!fields.containsKey(field.getName()))
+                    fields.put(field.getName(), field);
+            }
+        }
+        return fields;
+    }
+
+    private static <T> Map<String, PropertyDescriptor> scanProperties(Class<T> baseClass) {
+        BeanInfo beanInfo;
+        try {
+            beanInfo = Introspector.getBeanInfo(baseClass);
+        } catch (IntrospectionException e) {
+            throw Throwables.propagate(e);
+        }
+        Map<String, PropertyDescriptor> properties = new HashMap<String, PropertyDescriptor>();
+        for (PropertyDescriptor property : beanInfo.getPropertyDescriptors()) {
+            if (property.getName().equals("class"))
+                continue;
+            properties.put(property.getName(), property);
+        }
+        return properties;
+    }
+
+    static Map<Class<? extends Annotation>, Annotation> scanPropertyAnnotations(Field field, PropertyDescriptor property) {
+        Map<Class<? extends Annotation>, Annotation> annotations = new HashMap<Class<? extends Annotation>, Annotation>();
+        // annotations on getters should have precedence over annotations on fields
+        if (field != null)
+            scanFieldAnnotations(field, annotations);
+        Method getter = findGetter(property);
+        if (getter != null)
+            scanMethodAnnotations(getter, annotations);
+        return annotations;
+    }
+
+    private static Map<Class<? extends Annotation>, Annotation> scanFieldAnnotations(Field field, Map<Class<? extends Annotation>, Annotation> annotations) {
+        for (Annotation annotation : field.getAnnotations()) {
+            annotations.put(annotation.annotationType(), annotation);
+        }
+        return annotations;
     }
+
+    private static Map<Class<? extends Annotation>, Annotation> scanMethodAnnotations(Method method, Map<Class<? extends Annotation>, Annotation> annotations) {
```

Comment:
Ah nice, so this is where most of the magic happens to pull Annotations from parent classes/interfaces?